app/controllers/user_blocks_controller.rb

```diff
@@ -0,0 +1,78 @@
+class UserBlocksController < ApplicationController
+  before_action :check_sign_in_status
+  after_action :verify_authorized
+
+  def show
+    skip_authorization
+
+    if current_user.blocking?(params[:blocked_id].to_i)
+      render plain: "blocking"
+    else
+      render plain: "not-blocking"
+    end
+  end
+
+  def create
+
+    authorize UserBlock
+    @user_block = UserBlock.new(permitted_attributes(UserBlock))
+    @user_block.blocker_id = current_user.id
+    @user_block.config = "default"
+
+    if @user_block.save
+      block_chat_channel(@user_block.blocked_id)
+      current_user.stop_following(@user_block.blocked)
+      @user_block.blocked.stop_following(current_user)
+      render json: { outcome: "blocked" }
+    else
+      render json: { outcome: "error: #{@user_block&.errors&.full_messages}" }
```

Comment:
the error state should tell the HTTP client it's an error, so maybe here you should do something like:

```ruby
render json: { errors: @user_block&.errors&.full_messages }, status: :unprocessable_entity
```

Also, is there a scenario when `errors` is nil? If there are no errors `model.errors.full_messages` is an empty array, otherwise is an array of strings